ApollosProject/apollos-prototype
# Configurable Content Items

## DESCRIPTION
The old code was hard coded to assume that ContentItems were ContentChannelItems in Rock. This made functionality like "liking" hard to implement, even though it technically worked. This PR opens the door on the assumption that `__typenames` in our schema may not actually be ContentChannelItems. This should enable "liking" and "unliking" of things that aren't Content Items (but we still have some work to go to enable viewing all "Liked" content on a user's profile.

This also cleans up the configuration for ContentItems in the `config.yml` and opens the door for future enhancement. 

This PR also adds a function to easily extend all types a user has defined to make `sharing` and `liking` easy functionality to add/remove to exist ContentItem implementations. 

### What does this PR do, or why is it needed?

Described above.

### What design trade-offs/decisions were made?

- Rock Analytics on non content-items don't work perfectly yet
- Viewing non content items on the "Stuff I've liked" query doesn't work yet.

### How do I test this PR?

1. Run the app/api.
2. Like an item
 - It should change from unliked to liked
 - It should show up in your liked item list
3. Open a devotional item
 - it should show a split tab view with the bible
4. Open a media content item
  - you should be able to play the sermon. 

### What automated tests did you write?

Test covering new helper functions.

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
